src/drivers/github.js

```diff
@@ -288,6 +288,112 @@ class Github {
     });
   }
 
+  async pipelineJobs(opts = {}) {
+    const { ids } = opts;
+    const { owner, repo } = ownerRepo({ uri: this.repo });
+    const { actions } = octokit(this.token, this.repo);
+
+    const jobs = await Promise.all(
+      ids.map(async (id) => {
+        const { data } = await actions.getJobForWorkflowRun({
+          owner,
+          repo,
+          job_id: id
+        });
+
+        return data;
+      })
+    );
+
+    return jobs.map((job) => {
+      const { id, started_at: date, run_id: runId, status } = job;
+      return { id, date, runId, status };
+    });
+  }
+
+  async job(opts = {}) {
+    const { time, status = 'queued' } = opts;
+    const { owner, repo } = ownerRepo({ uri: this.repo });
+    const { actions } = octokit(this.token, this.repo);
+
+    const {
+      data: { workflow_runs: workflowRuns }
+    } = await actions.listWorkflowRunsForRepo({
+      owner,
+      repo,
+      status
+    });
+
+    let runJobs = await Promise.all(
+      workflowRuns.map(async (run) => {
+        const {
+          data: { jobs }
+        } = await actions.listJobsForWorkflowRun({
+          owner,
+          repo,
+          run_id: run.id,
+          status
+        });
+
+        return jobs;
+      })
+    );
+
+    runJobs = [].concat.apply([], runJobs).map((job) => {
+      const { id, started_at: date, run_id: runId } = job;
+      return { id, date, runId };
+    });
+
+    const job = runJobs.reduce((prev, curr) => {
+      const diffTime = (job) => Math.abs(new Date(job.date).getTime() - time);
+      return diffTime(curr) < diffTime(prev) ? curr : prev;
+    });
+
+    return job;
+  }
+
+  async pipelineRestart(opts = {}) {
+    const { jobId: job_id } = opts;
```

Comment:
Identifier 'job_id' is not in camel case

Reply:
Its not! Github uses only snake_case and I prefer variable assignment to conform the object

Reply:
> _I prefer variable assignment to conform the object_

The linter prefers otherwise, and it's going to yell the same each time you ask for an automated review. 🤷🏼‍♂️

Reply:
Ah ok I will add the lint exception,

Reply:
🙃  It it worth the exception? After @casperdcl's https://github.com/iterative/cml/pull/585#pullrequestreview-680822808, we chose not to upset the linter.

Reply:
Oh believe me we will have to add maaaaaaaaany lint inline exceptions. Not general exception but inline.
Thats pretty common in JS once you setup a style.

Reply:
> _Thats pretty common in JS once you setup a style._

...and you start to deviate from what it recommends. 😉 
***

Whatever! Mentioning on https://github.com/iterative/cml/issues/596 to keep track of style issues.

Reply:
> ...and you start to deviate from what it recommends.

Not really, if you review JS projects you will find tons of inline lint exceptions because of your style collides with others needs.

Reply:
Not using shorthand goes very strong against my declared OCD

Reply:
> Oh believe me we will have to add maaaaaaaaany lint inline exceptions

defeats the entire purpose of a linter. A linter is meant to ensure that code is consistently broken so that no humans ever waste time arguing about it.

Reply:
It's not because of the team's code @casperdcl is because s the third party code that one uses. Like here. Github uses snake case and if your style is not do shorthand you have to include exceptions. Whatever...

Reply:
And thats also the purpose of the inline exceptions

Reply:
We can always resort to the approach suggested on https://github.com/iterative/cml/pull/585#discussion_r649127382 to keep the linter happy without exceptions by converting identifiers to snake case only on the library function invocation.